x-pack/plugins/enterprise_search/server/routes/enterprise_search/indices.ts

```diff
@@ -291,6 +295,71 @@ export function registerIndexRoutes({
     })
   );
 
+  router.post(
+    {
+      path: '/internal/enterprise_search/indices/{indexName}/ml_inference/pipeline_processors',
+      validate: {
+        params: schema.object({
+          indexName: schema.string(),
+        }),
+        body: schema.object({
+          pipeline_name: schema.string(),
+          model_id: schema.string(),
+          source_field: schema.string(),
+          destination_field: schema.maybe(schema.nullable(schema.string())),
+        }),
+      },
+    },
+
+    elasticsearchErrorHandler(log, async (context, request, response) => {
+      const indexName = decodeURIComponent(request.params.indexName);
+      const { client } = (await context.core).elasticsearch;
+      const {
+        model_id: modelId,
+        pipeline_name: pipelineName,
+        source_field: sourceField,
+        destination_field: destinationField,
+      } = request.body;
+
+      try {
+        // Create the sub-pipeline for inference
```

Comment:
Since the `createPipelineResult` and `addSubPipelineToIndexSpecificMlPipeline` methods will always be called together here for the `POST` method, I'd recommend adding a single method to `x-pack/plugins/enterprise_search/server/utils/create_ml_inference_pipeline.ts` that handles this (and then, only have the route here call that one single method).

Reply:
Is it OK if I wrap the two steps in a single function (in `create_ml_inference_pipeline.ts`) rather than combining the code?
I still prefer to keep `createPipelineResult()` and `addSubPipelineToIndexSpecificMlPipeline()` separate for better cohesion and testability.

Reply:
Absolutely (and sorry, that is moreso what I meant, but didn't communicate it well) - yes, a new method that wraps these two steps in a single method.